simpleParser.go

```diff
@@ -367,6 +381,16 @@ func tokenIsChainOpType(token string) bool {
 	return token == TokenOperatorAnd || token == TokenOperatorAnd2 || token == TokenOperatorOr || token == TokenOperatorOr2
 }
 
+// Returns true if the value is to be used for pcre types
```

Comment:
I don't entirely follow the flow of the code here, but is this checking if the expression needs PCRE as opposed to standard Go regexp?

Reply:
That is correct. The idea is to try to use pcre only for lookahead/lookbehind cases, which are not supported by golang's regex library.

Reply:
Are you able to confirm that the behaviour of PCRE regexp and Golang regexp are identical except for lookback/lookahead support?

Reply:
I didn't test it myself but was reading up on https://en.wikipedia.org/wiki/RE2_(software) "Google's RE2 has a slightly smaller set of features than PCRE, but has very predictable run-time and a maximum memory allotment, making it suitable for use in server applications which require precise boundaries on memory usage and computational time. PCRE, on the other hand, has almost all of the features that a regular expression library can have, but has unpredictable run-time and memory usage and can grow unbounded."

And the syntax page: https://github.com/google/re2/wiki/Syntax for me shows that it is indeed a smaller subset of pcre.

Reply:
Good point. Will try removing pcre locally and see if I can get it to do that.

Reply:
Actually, transform.go already has:
```
	case PcreExpr:
		pcreRegex, err := pcre.Compile(expr.Pcre.(string), 0)
		if err != nil {
			return nil, errors.New("failed to compile PcreExpr: " + err.Message)
		}
		return NewFastVal(&pcreRegex), nil
```
But I can do something in parser too

Reply:
My concern was mainly around the failure mode, I just wanted to confirm that we were handling it appropriately.

Reply:
Actually, go doesn't even allow you to get the package if you don't have pcre installed:
```
administrator@administrator-VirtualBox:~/source/go/src/github.com/nelio2k/gojsonsm$ go get github.com/glenn-brown/golang-pkg-pcre/src/pkg/pcre
# github.com/glenn-brown/golang-pkg-pcre/src/pkg/pcre
../../glenn-brown/golang-pkg-pcre/src/pkg/pcre/pcre.go:52:18: fatal error: pcre.h: No such file or directory
compilation terminated.
```
and
```
administrator@administrator-VirtualBox:~/source/go/src/github.com/nelio2k/gojsonsm$ go build
# github.com/glenn-brown/golang-pkg-pcre/src/pkg/pcre
../../glenn-brown/golang-pkg-pcre/src/pkg/pcre/pcre.go:52:18: fatal error: pcre.h: No such file or directory
compilation terminated.
```
I think we should be safe here... even if people use this library independently, go will ask them to get the library and it'll do the pcre check automatically.

Reply:
Wont that block installation of gojsonsm if there is no system pcre library available?

Reply:
That sounds unfortunately right... I'm looking into ways to do ways to do C-styled macro... do you know anything like that in go?

Reply:
I believe the only way to do it would be to do it on the C side by having your own cgo wrapper around pcre.  I can look into it a bit more if you can't find any examples.

Reply:
You could also use go build flags to allow the user to specify the PCRE option which turns it on, and otherwise it would simply say that it wasn't compiled with that support.  This may work since most people don't need lookbehind/lookahead, but server builds could use the flag.

Reply:
Latest squashed PR has build tags built in